test/clickhousex/query_parse_test.exs

```diff
@@ -0,0 +1,35 @@
+defmodule Clickhousex.QueryParseTest do
+  use ExUnit.Case, async: true
+  alias Clickhousex.Query
+
+  describe "type" do
+    test "`INSERT INTO ... SELECT FROM ...` is marked as `:insert`" do
+      assert %Query{type: :insert} = Query.new("INSERT INTO table SELECT FROM other_table")
+    end
+
+    test "SELECT ... FROM ... WHERE column = 'Insert'` is marked as `:select`" do
+      assert %Query{type: :select} = Query.new("SELECT * FROM table WHERE column = 'Insert'")
+    end
+
+    test "`INSERT INTO ... VALUES ...` is marked as `:insert`" do
+      assert %Query{type: :insert} = Query.new("INSERT INTO table (*) VALUES (?, ?, ?)")
+    end
+
+    test "`CREATE TABLE ...` is marked as `:create`" do
+      statement = """
+      CREATE TABLE IF NOT EXISTS table_name
+      (name1 UInt16 NOT NULL) ENGINE = MergeTree
+      """
+
+      assert %Query{type: :create} = Query.new(statement)
+    end
+
+    test "`ALTER TABLE ... UPDATE` is marked as `:alter`" do
+      assert %Query{type: :alter} = Query.new("ALTER TABLE table UPDATE column1 = ?")
+    end
+
+    test "other statements are merked as `:update`" do
+      assert %Query{type: :update} = Query.new("SHOW DATABASES")
+    end
+  end
```

Comment:
I've added some basic tests here. I'll look into how `query.type` is actually used to make them more useful.

Reply:
yeah I'm not sure why it's significant...